src/dodal/devices/i24/beamstop.py

```diff
@@ -0,0 +1,37 @@
+from enum import Enum
+
+from ophyd_async.core import StandardReadable
+from ophyd_async.epics.motion import Motor
+from ophyd_async.epics.signal import epics_signal_rw
+
+
+class BSPositions(str, Enum):
+    CHECKBEAM = "CheckBeam"
+    DATACOLLECTION = "Data Collection"
+    DATACOLLECTIONFAR = "Data Collection Far"
+    TRAYMOUNT = "Tray Mount"
+    ROTATABLE = "Rotatable"
+    ROBOT = "Robot"
+
+
+class Beamstop(StandardReadable):
+    """Device to move the beamstop.
+
+    The positioner moves the x,y,z motors when a position is selected.
+    The additional rot_y motor is independent of the positioner and can to be moved
+    on its own as needed.
+
+    WARNING. Before moving the rot_y motor away from 0, it is important to make sure
+    that the backlight is in the "OUT" position to avoid a collision.
```

Comment:
Could: We should protect against this somehow. What we have done in other places where there are potential collisions is make a new composite device like `BeamStopAndBacklight` where moving `rot_y` would move the backlight first. Or we could not allow the rot_y to move at all in this case

Reply:
For a device, I would pick the second option, so check if the backlight is in and if it is don't allow rot_y to move (the beamstop can still go out of the way by moving the other motors with the positioner). 
However, I'd make a separate issue for creating this device and discuss it better with the scientists - I got the impression when I asked that it's not really used - have yet to take a good look at what happens in GDA though. For the serial stuff, they only set this to 0 at the start of the collection to make sure it won't be in the backlight's way.

Reply:
Issue [here](https://github.com/DiamondLightSource/dodal/issues/646)